static/src/javascripts/projects/common/modules/experiments/ab.js

```diff
@@ -0,0 +1,338 @@
+
+export const trackEvent = () => recordOphanAbEvent(getAbLoggableObject());
+
+/**
+ * Register a test and variant's complete state with Ophan
+ *
+ * @param test
+ * @param {string} variantId
+ * @param {boolean} complete
+ * @returns {Function} to fire the event
+ */
+const recordTestComplete = (
+    test: ABTest,
+    variantId: string,
+    complete: boolean
+) => {
+    const data = {};
+    const variant = abUtils.getVariant(test, variantId);
+
+    data[test.id] = abData(variantId, String(complete), variant.campaignCodes);
+
+    return () => recordOphanAbEvent(data);
+};
+
+// Finds variant in specific tests and runs it
+const runTest = test => {
+    if (abUtils.isParticipating(test) && testCanRunChecks.testCanBeRun(test)) {
+        const participations = abUtils.getParticipations();
+        const variantId = participations[test.id].variant;
+        const variant = abUtils.getVariant(test, variantId);
+
+        if (variant) {
+            variant.test();
+        } else if (!segmentUtil.isInTest(test) && test.notInTest) {
+            test.notInTest();
+        }
+    }
+};
+
+const allocateUserToTest = test => {
+    // Only allocate the user if the test is valid and they're not already participating.
+    if (testCanRunChecks.testCanBeRun(test) && !abUtils.isParticipating(test)) {
+        abUtils.addParticipation(test, segmentUtil.variantIdFor(test));
+    }
+};
+
+/**
+ * Create a function that sets up listener to fire an Ophan `complete` event. This is used in the `success` and
+ * `impression` properties of test variants to allow test authors to control when these events are sent out.
+ *
+ * @see {@link defersImpression}
+ * @param {Boolean} complete
+ * @returns {Function}
+ */
+const registerCompleteEvent = complete => test => {
+    const variantId = abUtils.getTestVariantId(test.id);
+
+    if (variantId && variantId !== 'notintest') {
+        const variant = abUtils.getVariant(test, variantId);
+        const listener =
+            (complete ? variant.success : variant.impression) || noop;
+
+        try {
+            listener(recordTestComplete(test, variantId, complete));
+        } catch (err) {
+            reportError(err, {}, false);
+        }
+    }
+};
+
+export const shouldRunTest = (id: string, variant: string) => {
+    const test = getTest(id);
+
+    return (
+        test &&
+        abUtils.isParticipating(test) &&
+        abUtils.getTestVariantId(id) === variant &&
+        testCanRunChecks.testCanBeRun(test)
+    );
+};
+
+const getForcedIntoTests = () => {
+    if (/^#ab/.test(window.location.hash)) {
+        const tokens = window.location.hash.replace('#ab-', '').split(',');
+
+        return tokens.map(token => {
+            const abParam = token.split('=');
+
+            return {
+                id: abParam[0],
+                variant: abParam[1],
+            };
+        });
+    }
+
+    return JSON.parse(local.get('gu.devtools.ab')) || [];
+};
+
+export const getTests = () => TESTS;
```

Comment:
now that these are exports, it looks a bit odd. Could they be exported in a specific way to show they are tests? Like https://github.com/guardian/frontend/blob/master/static/src/javascripts-legacy/projects/commercial/modules/article-body-adverts.js#L66